pkg/virt-handler/vm.go

```diff
@@ -3239,23 +3239,9 @@ func (d *VirtualMachineController) finalizeMigration(vmi *v1.VirtualMachineInsta
 		return fmt.Errorf("%s: %v", errorMessage, err)
 	}
 
-	if vmi.Status.CurrentCPUTopology != nil {
-		// handle vCPU plug/unplug
-		options := virtualMachineOptions(
-			nil,
-			0,
-			nil,
-			d.capabilities,
-			map[string]*containerdisk.DiskInfo{},
-			d.clusterConfig)
-
-		if err := client.SyncVirtualMachineCPUs(vmi, options); err != nil {
-			log.Log.Object(vmi).Reason(err).Error(errorMessage)
-			d.recorder.Event(vmi, k8sv1.EventTypeWarning, err.Error(), "failed to change vCPUs")
-		} else {
-			controller.NewVirtualMachineInstanceConditionManager().RemoveCondition(vmi, v1.VirtualMachineInstanceVCPUChange)
-			vmi.Status.CurrentCPUTopology = nil
-		}
+	if err := d.hotplugCPU(vmi, client); err != nil {
```

Comment:
Having the hotplug here doesn't guarantee this will be retried. We need to move this into handlers sync.

Reply:
Following my conversation with @vladikr we would like to do it once rather than reconcile on it

Reply:
Mainly because if this command fails, I don't see what will we achieve by endlessly retrying. 
This should just never happen. If it did happen, then we need to finalize and record an event for the user to know that the hot plug has failed.